### PR TITLE
Avoid bind method

### PR DESCRIPTION
**Pull request by regislebrun**
+Changed the method binding to speed-up computations in DistributionImplementation:
Switched from the use of the general method binding to a specific binding to speed-up the computation of conditional CDF, following a very good idea of Denis Barbier.
+Made the getShiftedMoment() method visible in Distribution
+Added a vectorized version of some methods in Distribution
It concerns the following methods: computeConditionalPDF(), computeConditionalCDF() and computeConditionalQuantile().
The key point is to reuse the marginal distributions involved in these computations, as there extraction can be costly. These versions should be overloaded on a specific multivariate distribution/copula basis, not yet done.
+Improved the documentation of Distribution
Added the documentation of the getShiftedMoment() method, fixed some typos and the description of standard moments which was wrong.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -35,9 +35,10 @@
 === Python module ===
 
 === Miscellaneous ===
- * Improved the computeCDF() method of Normal 
- * Add HarrisonMcCabe, BreuschPagan and DurbinWatson tests to test homoskedasticity, autocorrelation of linear regression residuals 
+ * Improved the computeCDF() method of Normal
+ * Add HarrisonMcCabe, BreuschPagan and DurbinWatson tests to test homoskedasticity, autocorrelation of linear regression residuals
  * Add two samples Kolmogorov test
+ * Improved the speed of many algrithms based on method binding
```
Comment (jschueller) [inline ChangeLog]: algrithms

File: lib/src/Base/Algo/IteratedQuadrature.cxx [hunk 2]
```diff
@@ -21,7 +21,6 @@
 #include "openturns/IteratedQuadrature.hxx"
 #include "openturns/GaussKronrod.hxx"
 #include "openturns/Exception.hxx"
-#include "openturns/SpecFunc.hxx"
 #include "openturns/PersistentObjectFactory.hxx"
 #include "openturns/MethodBoundNumericalMathEvaluationImplementation.hxx"
```
Comment (jschueller) [inline lib/src/Base/Algo/IteratedQuadrature.cxx]: unneeded ?

File: lib/src/Uncertainty/Model/DistributionImplementation.cxx [hunk 3]
```diff
@@ -1602,10 +1602,10 @@ NumericalScalar DistributionImplementation::computeConditionalCDF(const Numerica
   const NumericalScalar xMax(conditionedDistribution->getRange().getUpperBound()[conditioningDimension]);
   if (x >= xMax) return 1.0;
   // Numerical integration with respect to x
-  const ConditionalPDFWrapper wrapper(y, conditionedDistribution);
-  const NumericalMathFunction f(bindMethod<ConditionalPDFWrapper, NumericalPoint, NumericalPoint>(wrapper, &ConditionalPDFWrapper::computeConditionalPDF, 1, 1));
```
Comment (jschueller) [inline lib/src/Uncertainty/Model/DistributionImplementation.cxx]: MethodBoundZZZ.hxx include is now useless

**Comment by jschueller** [comment]
do you plan to cure randommixture too ? its windows test is superslow too.

**Comment by regislebrun** [comment]
@jschueller I plan to replace all the calls to bindMethod() by a specific wrapper, but have no time to do the whole change now. As the current work already speed up many computations, in particular those related to the generic implementation of Rosenblatt's transformation, I propose to integrate these changes as a first step toward a total replacement of bindMethod. I will have a look at RandomMixture, but in a separate PR.

**Comment by jschueller** [comment]
@regislebrun, ok

**Comment by dbarbier** [comment]
+1, I do not have time to fully review these changes now, but they look great.

File: lib/src/Uncertainty/Model/DistributionImplementation.cxx (hunk 3 above)
Comment (regislebrun) [inline lib/src/Uncertainty/Model/DistributionImplementation.cxx]: @jschueller No: some other methods still use bindMethod(), such as computeMinimumVolumeInterval()

File: lib/src/Uncertainty/Model/openturns/DistributionImplementation.hxx [hunk 4]
```diff
@@ -726,66 +728,206 @@ protected:
...
+    }
+
+    UnsignedInteger getOutputDimension() const
+    {
+      return 1;
+    }
+
+    Description getInputDescription() const
+    {
+      return Description::BuildDefault(2, "x");;
```
Comment (jschueller) [inline lib/src/Uncertainty/Model/openturns/DistributionImplementation.hxx]: ;;)

**Comment by jschueller** [comment]
+1

**Pull request merged**